static/js/operation.js

```diff
@@ -92,30 +96,181 @@ function parseTable(text, data) {
+    // multiplied gc counts by 6 to convert decimals into integers
+    switch (line.charAt(i).toUpperCase()) {
+      case 'G':
+      case 'C':
+      case 'S':
+        count += 6;
+        break;
+      case 'R':
+      case 'Y':
+      case 'K':
+      case 'M':
+      case 'N':
+        count += 3;
+        break;
+      case 'D':
+      case 'H':
+        count += 2;
+        break;
+      case 'B':
+      case 'V':
+        count += 4;
+    }
+  }
+
+  // reconverting gc counter into decimals
+  count /= 6;
+  return count;  
+}
+
+
+/**
+ * Gets the file format of the input data
+ * @function getFileFormat
+ * @param {String} text - multi-line string in tsv format
+ * @returns {String} - file type of input data
+ */
+function getFileFormat(text) {
+  // searching for unique starting sequences of different file formats
+  var spades_regex = /^>NODE_\d+\_length_\d+\_cov_\d*\.?\d*/g;
+  var megahit_regex = /^>k\d+_\d+\sflag=\d+\smulti=\d*\.?\d*\slen=\d+/g;
+  if (text.search(spades_regex) === 0) {
+    return 'spades';
+  }
+  else if (text.search(megahit_regex) === 0) {
+    return 'megahit';
+  }
+  return null;
+}
+
+
+/**
+ * Parses and retrieves information in the contig titles of the input data
+ * @function parseContigTitles
+ * @param {String} line - one line string of the contig title
+ * @param {String} format - file type of input data
+ * @returns {Array.<String, String, String>} - id, length, coverage of contig
+ */
+function parseContigTitles(line, format) {
+  var id = '';
+  var length = 0;
+  var coverage = 0;
+  if (format === 'spades') {
+    var regex = /(?<=_)(\+|-)?[0-9]*(\.[0-9]*)?$|\d+/g;
+    [id, length, coverage] = line.match(regex);
+    return [id, coverage];
+  }
+  else if (format === 'megahit') {
+    var regex = /(?<==|_)[0-9]*(\.[0-9]*)?/g;
+    [id, , coverage, length] = line.match(regex);
+    return [id, coverage];
+  }
+  return null;
+}
+
+
+/**
+ * Formats data into usable types
+ * @function formatData
+ * @param {Object} data - data object
+ * @param {Matrix} dataframe - data points for available characteristices of contigs
+ * @param {Object} columns - available characteristics of contigs
+ * @returns {Array.<Object, Object, Object>} - id, length, coverage of contig
+ */
+function formatData(data, dataframe, columns) {
   var deci = {};
   var cats = {};
   var feats = {};
 
   // identify field types and re-format data
   var types = ['id'];
-  for (var i = 1; i < cols.length; i++) {
+  for (var i = 1; i < columns.length; i++) {
     var arr = [];
-    for (var j = 0; j < df.length; j++) {
-      arr.push(df[j][i]);
-    }
-    var x = parseFieldType(cols[i], arr);
-    var type = x[0];
-    var col = x[1];
-    types.push(type); // identified type
-    cols[i] = col; // updated name
-    for (var j = 0; j < df.length; j++) {
-      df[j][i] = arr[j];
+    for (var j = 0; j < dataframe.length; j++) {
+      arr.push(dataframe[dataframe.length - j - 1][i]);
```

Comment:
@AbhinavChede When I was coding I noticed that this line of code turns the data frame around. Maybe I missed something here, but could you let me know your original rationale of editing `[j]` as `[df.length - j - 1]`?

Reply:
@qiyunzhu I changed that line because at first I was using the formatData method for assemblies as well. And when I used the formatData in its original version it was flipping the assemblies. But now since the parseAssembly function has no dependency on the formatData method I think we can revert it to its original version. Also on a side note, I printed (after the dependency was removed) the df for tables and for both ways (`[j]` and `[df.length - j - 1]`)  the df of the table was printing normally and not backwards. So, I guess later in the pipeline the df is being restructured properly.

Reply:
@AbhinavChede Thank you for letting me know! That makes sense. I guess that at a later point the data table was sorted by contig ID prior to printing. I couldn't recall where though. The codebase is bit big...